modules/oe_content_featured_media_field/tests/src/FunctionalJavascript/FeaturedMediaFieldWidgetTest.php

```diff
@@ -0,0 +1,198 @@
+   *
+   * @var \Drupal\field\Entity\FieldConfig
+   */
+  protected $field;
+
+  /**
+   * The media entity used in this test class.
+   *
+   * @var \Drupal\media\Entity\Media
+   */
+  protected $mediaEntity;
+
+  /**
+   * The display options to use for the widget.
+   *
+   * @var array
+   */
+  protected $displayOptions = [
+    'type' => 'oe_featured_media_widget',
+    'settings' => [
+      'match_operator' => 'CONTAINS',
+      'match_limit' => 10,
+      'size' => 60,
+    ],
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'node',
+    'field',
+    'field_ui',
+    'media',
+    'oe_media',
+    'oe_content_featured_media_field',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+
+    $this->createContentType(['type' => 'page']);
+
+    $user = $this->drupalCreateUser([
+      'access content',
+      'create page content',
+    ]);
+    $this->drupalLogin($user);
+
+    $this->fieldStorage = FieldStorageConfig::create([
+      'field_name' => 'featured_media_field',
+      'entity_type' => 'node',
+      'type' => 'oe_featured_media',
+      'cardinality' => 1,
+      'entity_types' => ['node'],
+    ]);
+    $this->fieldStorage->save();
+
+    $this->field = FieldConfig::create([
+      'label' => 'Featured media field',
+      'field_name' => 'featured_media_field',
+      'entity_type' => 'node',
+      'bundle' => 'page',
+      'settings' => [
+        'handler' => 'default:media',
+        'handler_settings' => [
+          'target_bundles' => [
+            'image' => 'image',
+          ],
+        ],
+        'sort' => [
+          'field' => '_none',
+        ],
+        'auto_create' => '0',
+      ],
+      'required' => FALSE,
+    ]);
+    $this->field->save();
+
+    // Create an image file.
+    \Drupal::service('file_system')->copy($this->root . '/core/misc/druplicon.png', 'public://example.jpg');
+    $image = File::create(['uri' => 'public://example.jpg']);
+    $image->save();
+
+    // Create a media entity of image media type.
+    $media_type = $this->container->get('entity_type.manager')->getStorage('media_type')->load('image');
+    $this->mediaEntity = Media::create([
+      'bundle' => $media_type->id(),
+      'name' => 'Test image',
+      'field_media_image' => [
+        [
+          'target_id' => $image->id(),
+          'alt' => 'default alt',
+          'title' => 'default title',
+        ],
+      ],
+    ]);
+    $this->mediaEntity->save();
+
+    // Prepare the default view display for rendering.
+    $display = \Drupal::service('entity_display.repository')
+      ->getFormDisplay($this->field->getTargetEntityTypeId(), $this->field->getTargetBundle())
+      ->setComponent($this->fieldStorage->getName(), $this->displayOptions);
+    $display->save();
+  }
+
+  /**
+   * Tests the featured media widget.
+   */
+  public function testFeaturedMediaWidget(): void {
+    // Visit the node add page.
```

Comment:
Test empty case.

Reply:
It is tested below.

Reply:
Where? I don't see where you save the node with or without values and then asserting the saved values.

Reply:
Here we are testing the behavior of the widget, not the field storage. That is tested in the Kernel test.